modules/core/src/mathfuncs_core.simd.hpp

```diff
@@ -725,7 +725,7 @@ void log32f( const float *_x, float *y, int n )
 
         yf0 = v_fma(v_cvt_f32(yi0), vln2, yf0);
 
-        v_float32 delta = v_reinterpret_as_f32(h0 == vx_setall_s32(510)) & vshift;
+        v_float32 delta = v_select(v_reinterpret_as_f32(h0 == vx_setall_s32(510)), vshift, vx_setall<float>(0));
```

Comment:
There is a possibility that this version will be slower on SSE2 and SSE3. However I think the line could be optimized by a compiler.